packages/block-editor/src/components/use-paste-styles/index.js

```diff
@@ -36,31 +36,36 @@ function looksLikeBlocks( text ) {
 	}
 }
 
+/**
+ * Style attributes are attributes being added in `block-editor/src/hooks/*`.
+ * (Except for some unrelated to style like `anchor` or `settings`.)
+ * They generally represent the default block supports.
+ */
+const STYLE_ATTRIBUTES = [
+	'align',
+	'borderColor',
+	'backgroundColor',
+	'textColor',
+	'gradient',
+	'className',
+	'fontFamily',
+	'fontSize',
+	'layout',
+	'style',
+];
+
 /**
  * Get the "style attributes" from a given block.
- * A "style attribute" is an attribute that doesn't have `__experimentalRole` of `content`.
  *
  * @param {WPBlock} block The input block.
  * @return {Object} the filtered attributes object.
  */
 function getStyleAttributes( block ) {
-	const blockType = getBlockType( block.name );
-	const attributes = {};
-	for ( const [ attribute, attributeType ] of Object.entries(
-		blockType.attributes
-	) ) {
-		// Mark every attribute that isn't "content" as a style attribute.
-		if ( attributeType.__experimentalRole !== 'content' ) {
-			// Apply all attributes even when they are undefined to allow overriding styles.
-			attributes[ attribute ] = Object.hasOwn(
-				block.attributes,
-				attribute
-			)
-				? block.attributes[ attribute ]
-				: attributeType.default;
-		}
-	}
-	return attributes;
+	// Override attributes that are not present in the block to their defaults.
+	return STYLE_ATTRIBUTES.reduce( ( attributes, attributeKey ) => {
```

Comment:
@ntsekouras Out of interest, what issues do you see this causing? The image block doesn't use `align` block support, but still supports the same values as blocks that do, so I'm not sure this would be a problem.

I think there's probably a risk of causing weird issues with plugins that add attributes of the same name, but work differently. But I think block supports features generally clobber those attributes anyway, essentially they're reserved attribute names (and should probably be documented as such).

Reply:
I think there are more core blocks with some such attributes, but doesn't really matter, because as you say there could be any block(core or third party) that have same attribute names but do something different. We already have `hasBlockSupport` selector for that and it's better to be explicit than leave things a bit more loose.

Reply:
Fair enough. I think it'll be a challenge deriving the support name from the attribute name. That may require some refactoring of the block support hooks so that it can be reverse engineered. The only way I can see it'd currently be possible is to call the various `addAttribute` functions and then inspect the result.

Reply:
> I think it'll be a challenge deriving the support name from the attribute name

The list of attributes here is arbitrary, so couldn't we also have an arbitrary mapping of those to their respective block support to check before overriding the attribute?

Reply:
I've been trying to get the various `has*Support` from the hooks folder. However, every time I import from any of the modules it throws an error during runtime. I think it's because of a circular dependencies issue. Unfortunately, fixing the circular dependencies requires a large refactor. I might just try to duplicate the code here and see if anyone has better ideas.

Reply:
Added in [5526d1b](https://github.com/WordPress/gutenberg/pull/45477/commits/5526d1ba3a6709d96fba94c6220038e405d491c5).